tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -1619,6 +1619,18 @@ def map(self, map_func, num_parallel_calls=None, deterministic=None):
     >>> list(d.as_numpy_iterator())
     [b'HELLO', b'WORLD']
 
+    3) Use `tf.numpy_function`, which allows you to write arbitrary python
+    code. Note here that `tf.py_function` accepts `tf.Tensor` whereas
+    `tf.numpy_function` accepts numpy arrays. For example:
```

Comment:
`tf.numpy_function` only accepts numpu arrays and only returns numpy arrays

https://www.tensorflow.org/api_docs/python/tf/numpy_function

Reply:
done